src/extension.ts

```diff
@@ -68,10 +68,29 @@ export function activate(context: vscode.ExtensionContext) {
     }
   });
 
+  let nbUpdate: number = 0;
   vscode.tasks.onDidEndTask((event) => {
     const taskName = event.execution.task.name;
     if (taskName.startsWith("Update Container")) {
-      containerManager.manageContainer();
+      if (containerManager.containerStatus() == true) {
```

Comment:
I suggest we create a method of `containerManager` that will perform the stuff you've added (the retries etc) so all the container logic stays in the same class

Reply:
Done